.github/workflows/cpp.yml

```diff
@@ -55,25 +55,32 @@ jobs:
         addlicense -check .
 
   build:
+    name: build (${{ matrix.runtime }})
+
     runs-on: ubuntu-latest
 
     strategy:
+      fail-fast: false
       matrix:
-        # TODO(mathetake): Add other runtimes.
-        runtime: [ "wasmtime" ]
+        runtime: ["wamr", "wasmtime", "wavm"]
 
     steps:
     - uses: actions/checkout@v2
 
-    - name: Mount bazel cache
-      uses: actions/cache@v1
+    - name: Install dependency
+      run: sudo apt-get install ninja-build
+
+    - name: Mount Bazel cache
+      uses: actions/cache@v2
       with:
-        path: "/home/runner/.cache/bazel"
-        key: bazel-${{ matrix.runtime }}
+        path: |
+          ~/.cache/bazel
+          ~/.cache/bazelisk
+        key: bazel-${{ hashFiles('WORKSPACE', '.bazelrc', '.bazelversion', 'bazel/cargo/Cargo.raze.lock', 'bazel/dependencies.bzl', 'bazel/repositories.bzl') }}-${{ matrix.runtime }}
 
     - name: Test
       run: |
-        bazel test --define runtime=${{ matrix.runtime }} //...
+        bazel test --define runtime=${{ matrix.runtime }} //test/...
 
     - name: Test (signed Wasm module)
       run: |
```

Comment:
Oops! Because `--cxxopt` is used to provide the public key to verify signed Wasm modules, everything is rebuilt twice because of the flags change, which also invalidates cache.

I think we have to add another dimension to the matrix for signed/unsigned Wasm modules.

Sorry about that!

Reply:
Yeah, cache is definitely busted.

Reply:
OK, problem solved!

Reply:
Thank you!